docs/sequence/error_scenarios/PISPLinkingRegisterCredentialPhase.puml

```diff
@@ -0,0 +1,297 @@
+rnote right of DFSP_TP_IN: state: signedConsentRecieved
+DFSP_TP_IN -> DFSP_LM: model.validateSignedConsent()
+activate DFSP_LM
+rnote right of DFSP_TP_IN: state: pendingRegistrationAndValidation
+DFSP_TP_IN --> Switch: 202 Accepted
+deactivate Switch
+deactivate DFSP_TP_IN
+DFSP_LM -> DFSP_LM: Check signed consent.
+rnote right of DFSP_LM #LightBlue
+Does this need a backend request or can the scheme adapter check the consent.
+end note
+rnote right of DFSP_LM: state: signedConsentChecked
+DFSP_LM-> DFSP_LM: model.validateWithAuthService()
+DFSP_LM -> Switch: POST /consents/1a2b3c4d
+activate Switch
+
+rnote left of DFSP_LM #LightBlue
+POST /consents/1a2b3c4d
+FSIOP-Source: dfspa
+FSIOP-Destination: central-auth
+{
+  scopes: [{
+    accountId: 'dfspa.username.1234',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+    accountId: 'dfspa.username.5678',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+  }],
+  credential: {
+    credentialType: "FIDO",
+    status: "PENDING",
+    payload: PublicKeyCredential
+  }
+}
+end note
+Switch --> DFSP_LM: 202 Accepted
+DFSP_LM -> DFSP_LM: this.saveToKVS()
+deactivate DFSP_LM
+
+Switch -> AUTH: POST /consents/1a2b3c4d
+activate AUTH
+AUTH --> Switch: 202 Accepted
+AUTH -> AUTH: Check consent.
+AUTH -> Switch: PUT /consents/1a2b3c4d
+
+rnote left of AUTH #LightBlue
+PUT /consents/1a2b3c4d
+FSIOP-Source: central-auth
+FSIOP-Destination: dfspa
+{
+  scopes: [{
+    accountId: 'dfspa.username.1234',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+    accountId: 'dfspa.username.5678',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+  }],
+  credential: {
+    credentialType: "FIDO",
+    status: "VERIFIED",
+    payload: PublicKeyCredential
+  }
+}
+end note
+
+Switch --> AUTH: 200 OK
+Switch -> DFSP_TP_IN: PUT /consents/1a2b3c4d
+activate DFSP_TP_IN
+DFSP_TP_IN -> DFSP_TP_IN: const model = await loadFromKVS({key: 1a2b3c4d})
+DFSP_TP_IN -> DFSP_TP_IN: consentResponseRecieved()
+rnote right of DFSP_TP_IN: state: consentResponseRecieved
+DFSP_TP_IN -> DFSP_LM: Auth Service response recieved
+activate DFSP_LM
+DFSP_TP_IN --> Switch: 200 OK
+deactivate Switch
+deactivate DFSP_TP_IN
+
+AUTH -> ALS: POST /participants/CONSENTS/1a2b3c4d
+activate ALS
+ALS --> AUTH: 202 Accepted
+deactivate AUTH
+
+ALS -> DFSP_TP_IN: PUT /participants/CONSENTS/1a2b3c4d
+activate DFSP_TP_IN
+DFSP_TP_IN -> DFSP_TP_IN: const model = await loadFromKVS({key: 1a2b3c4d})
+DFSP_TP_IN -> DFSP_TP_IN: participantResponseRecieved()
+rnote right of DFSP_TP_IN: state: participantsResponseRecieved
+DFSP_TP_IN -> DFSP_LM: Participant response recieved
+DFSP_TP_IN --> ALS: 200 Accepted
+deactivate ALS
+deactivate DFSP_TP_IN
+rnote right of DFSP_LM: state: consentRegisteredAndValidated
+DFSP_LM -> DFSP_LM: model.finalizeConsent()
+
+loop for each scope in ""Consents.scopes""
+DFSP_LM -> Switch: POST /participants/THIRD_PARTY_LINK/dfsp.username.5678
```

Comment:
Hmm what if one of the `THIRD_PARTY_LINK` requests fails, but the other doesn't...

Reply:
```
  errorInformation: {
    errorCode: '7214',
    errorDescription: 'Failed to register account links with oracle'
  }
```

The requests should be wrapped in Promise.all(). If one times out I guess we consider it a complete failure for now?
Idk...how we would handle partial successes...thats going to be confusing. If that IS a scenario, lets leave that for a follow up.

Reply:
Yeah, leave that to me for now... I'll need to discuss with Michael